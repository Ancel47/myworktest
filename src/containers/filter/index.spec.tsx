import React from "react";

import {
    render,
    TestWrapper,
    MockJSONServer,
    getByTestId,
    fireEvent,
} from "@test";
import { act } from "react-dom/test-utils";
import { Form, FormItem, SelectInput, TextInput } from "@components";
import { Filter } from "./";

describe("containers / filter", () => {
    const filters = (
        <Form
            data-testid="filter-form"
            name="filter-form"
            style={{
                marginBlock: 10,
            }}
            layout="inline"
        >
            <FormItem label="Search" name="q" data-testid="search">
                <TextInput placeholder="Search" />
            </FormItem>
            <FormItem label="Status" name="status" hidden data-testid="status">
                <SelectInput
                    allowClear
                    placeholder="All Status"
                    options={[
                        {
                            label: "Active",
                            value: "active",
                        },
                        {
                            label: "Draft",
                            value: "draft",
                        },
                    ]}
                />
            </FormItem>
        </Form>
    );

    it("render correct filters", async () => {
        const { container, findByTestId } = render(
            <Filter resourceName="posts">{filters}</Filter>,
            {
                wrapper: TestWrapper({
                    dataProvider: MockJSONServer,
                    resources: [],
                }),
            },
        );

        expect(getByTestId(container, "search"));
        expect(await getByTestId(container, "status"));
        expect(
            await getByTestId(container, "status").getAttribute("class"),
        ).toContain("ant-form-item-hidden");

        expect(await findByTestId("filter-form")).toMatchSnapshot();
    });

    it("render correct filters dropdown", async () => {
        const { container, findByTestId } = render(
            <Filter resourceName="posts">{filters}</Filter>,
            {
                wrapper: TestWrapper({
                    dataProvider: MockJSONServer,
                    resources: [],
                }),
            },
        );

        act(() => {
            fireEvent.click(getByTestId(container, "filters-dropdown-button"));
        });

        expect(await findByTestId("filters-dropdown")).toMatchSnapshot();
    });
});
