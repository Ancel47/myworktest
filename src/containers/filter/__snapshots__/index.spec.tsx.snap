// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`containers / filter render correct filters 1`] = `
<form
  class="ant-form ant-form-inline"
  data-testid="filter-form"
  id="filter-form"
  style="margin-block: 10px;"
>
  <div
    class="ant-row ant-form-item"
    data-testid="search"
  >
    <div
      class="ant-col ant-form-item-label"
    >
      <label
        class=""
        for="filter-form_q"
        title="Search"
      >
        Search
      </label>
    </div>
    <div
      class="ant-col ant-form-item-control"
    >
      <div
        class="ant-form-item-control-input"
      >
        <div
          class="ant-form-item-control-input-content"
        >
          <input
            class="ant-input"
            id="filter-form_q"
            placeholder="Search"
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
  </div>
  <div
    class="ant-row ant-form-item ant-form-item-hidden"
    data-testid="status"
  >
    <div
      class="ant-col ant-form-item-label"
    >
      <label
        class=""
        for="filter-form_status"
        title="Status"
      >
        Status
      </label>
    </div>
    <div
      class="ant-col ant-form-item-control"
    >
      <div
        class="ant-form-item-control-input"
      >
        <div
          class="ant-form-item-control-input-content"
        >
          <div
            class="ant-select ant-select-single ant-select-allow-clear ant-select-show-arrow"
          >
            <div
              class="ant-select-selector"
            >
              <span
                class="ant-select-selection-search"
              >
                <input
                  aria-activedescendant="filter-form_status_list_0"
                  aria-autocomplete="list"
                  aria-controls="filter-form_status_list"
                  aria-haspopup="listbox"
                  aria-owns="filter-form_status_list"
                  autocomplete="off"
                  class="ant-select-selection-search-input"
                  id="filter-form_status"
                  readonly=""
                  role="combobox"
                  style="opacity: 0;"
                  type="search"
                  unselectable="on"
                  value=""
                />
              </span>
              <span
                class="ant-select-selection-placeholder"
              >
                All Status
              </span>
            </div>
            <span
              aria-hidden="true"
              class="ant-select-arrow"
              style="user-select: none;"
              unselectable="on"
            >
              <span
                aria-label="down"
                class="anticon anticon-down ant-select-suffix"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  data-icon="down"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                  />
                </svg>
              </span>
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</form>
`;

exports[`containers / filter render correct filters dropdown 1`] = `
<ul
  class="ant-dropdown-menu ant-dropdown-menu-light ant-dropdown-menu-root ant-dropdown-menu-vertical"
  data-testid="filters-dropdown"
  role="menu"
  tabindex="0"
>
  <li
    class="ant-dropdown-menu-item ant-dropdown-menu-item-only-child"
    role="menuitem"
  >
    Search
  </li>
  <li
    class="ant-dropdown-menu-item ant-dropdown-menu-item-only-child"
    role="menuitem"
  >
    Status
  </li>
</ul>
`;
